ext/OceananigansAMDGPUExt.jl

```diff
@@ -30,6 +35,19 @@ on_architecture(::ROCGPU, a::StepRangeLen) = a
 @inline convert_to_device(::ROCGPU, args) = AMDGPU.rocconvert(args)
 @inline convert_to_device(::ROCGPU, args::Tuple) = map(AMDGPU.rocconvert, args)
 
+function plan_forward_transform(A::ROCArray, ::Union{Bounded, Periodic}, dims, planner_flag)
+    length(dims) == 0 && return nothing
+    return AMDGPU.rocFFT.plan_fft!(A, dims)
+end
+
+function plan_backward_transform(A::ROCArray, ::Union{Bounded, Periodic}, dims, planner_flag)
+    length(dims) == 0 && return nothing
+    return AMDGPU.rocFFT.plan_bfft!(A, dims)
```

Comment:
this is the `plan_bfft!`, not the `plan_ifft!`
do we need to add a normalisation or something?